synapse/lib/types.py

```diff
@@ -760,7 +763,7 @@ def _initEdgeBase(self, n1, n2):
 
         if self.n2forms is not None:
             if n2[0] not in self.n2forms:
-                raise s_exc.BadTypeValu(valu, mesg='Invalid dest node for edge type: %r' % n2[0])
+                raise s_exc.BadTypeValu(valu=n2[0], mesg='Invalid dest node for edge type')
```

Comment:
Do we need to check that `n1` and `n2` aren't `None` here to avoid potential exception inside the raise here?

Reply:
`n1` and `n2` are already ndef values, so if they had failed to norm up front then we would've seen a BadTypeError raised by the previous `norm()` calls.